src/path.c

```diff
@@ -782,7 +782,8 @@ int git_path_resolve_relative(git_buf *path, size_t ceiling)
 
 int git_path_apply_relative(git_buf *target, const char *relpath)
 {
-	git_buf_joinpath(target, git_buf_cstr(target), relpath);
+	if (git_buf_joinpath(target, git_buf_cstr(target), relpath) < 0)
```

Comment:
This should not be necessary.  `git_path_resolve_relative` [checks for an OOM on `target`](https://github.com/libgit2/libgit2/blob/594a5d12d4b9ba3ea542722805ada1028bd1b646/src/path.c#L708) as its first check...